modules/shapes/assets/js/frontend/handlers/text-path.js

```diff
@@ -0,0 +1,217 @@
+export default class WordArtHandler extends elementorModules.frontend.handlers.Base {
+	getDefaultSettings() {
+		return {
+			selectors: {
+				pathContainer: '.elementor-text-path',
+				svg: '.elementor-text-path > svg',
+			},
+		};
+	}
+
+	getDefaultElements() {
+		const { selectors } = this.getSettings();
+		const element = this.$element[ 0 ];
+
+		return {
+			widgetWrapper: element,
+			pathContainer: element.querySelector( selectors.pathContainer ),
+			svg: element.querySelector( selectors.svg ),
+			textPath: element.querySelector( selectors.textPath ),
+		};
+	}
+
+	/**
+	 * Initialize the object.
+	 *
+	 * @returns {void}
+	 */
+	onInit() {
+		this.elements = this.getDefaultElements();
+
+		// Generate unique IDs using the wrapper's `data-id`.
+		this.pathId = `elementor-path-${ this.elements.widgetWrapper.dataset.id }`;
+		this.textPathId = `elementor-text-path-${ this.elements.widgetWrapper.dataset.id }`;
+
+		if ( ! this.elements.svg ) {
+			return;
+		}
+
+		this.initTextPath();
+	}
+
+	/**
+	 * Set the start offset for the text.
+	 *
+	 * @param offset {string|int} The text start offset.
+	 *
+	 * @returns {void}
+	 */
+	setOffset( offset ) {
+		if ( ! this.elements.textPath ) {
+			return;
+		}
+
+		if ( this.isRTL() ) {
+			offset = 100 - parseInt( offset );
+		}
+
+		this.elements.textPath.setAttribute( 'startOffset', offset + '%' );
+	}
+
+	/**
+	 * Handle element settings changes.
+	 *
+	 * @param setting {Object} The settings object from the editor.
+	 *
+	 * @returns {void}
+	 */
+	onElementChange( setting ) {
+		const {
+			start_point: startPoint,
+			text,
+		} = this.getElementSettings();
+
+		switch ( setting ) {
+			case 'start_point':
+				this.setOffset( startPoint.size );
+				break;
+
+			case 'text':
+				this.setText( text );
+				break;
+
+			case 'text_path_direction':
+				this.setOffset( startPoint.size );
+				this.setText( text );
+				break;
+
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * Attach a unique id to the path.
+	 *
+	 * @returns {void}
+	 */
+	attachIdToPath() {
+		const path = this.elements.svg.querySelector( '[data-path-anchor]' ) || this.elements.svg.querySelector( 'path' );
```

Comment:
```suggestion
		const path = this.elements.svg.querySelector( '[data-path-anchor], path' );
```

Reply:
Already tried that in the previous review, it doesn't work as expected.
I need to prioritize the `data-path-anchor` over the `path`.

See this fiddle:
https://jsfiddle.net/kb23x95t/

Reply:
OK, Please explain in code.